src/app/features/backoffice/pages/activity-form/activity-form.component.ts

```diff
@@ -95,24 +97,55 @@ export class ActivityFormComponent implements OnInit {
     }
   }
 
-  ngOnInit(): void {
+  async ngOnInit(): Promise<void> {
     if (typeof this.route.snapshot.params["idActivity"] !== "undefined") {
       this.editing = true;
       this.action = "Edit activity";
       const url: string =
-        environment.apiUrl +
-        "/activities/" +
-        this.route.snapshot.params["idActivity"];
-      this.httpService.get(url).subscribe((result) => {
-        let resultData: any = JSON.parse(JSON.stringify(result));
-        this.anActivity = JSON.parse(JSON.stringify(resultData.data));
+        environment.apiUrl +"/activities/";     
+        const req=  this.privateBackofficeService.getActivityById(url,this.route.snapshot.params["idActivity"]);        
+        req.then(response => { 
+          let resultData: any = response;
```

Comment:
No usar any, aplicar la interfaz del modelo

Reply:
Aca sucede que si uso la entidad (linea 2)
1- let resultData: any = response;
 2-this.anActivity = resultData.data;
 Pero no modelo en Activity los campos 
  success
  data
  message
  Me parece inecesario modelar eso, ya que si tendria que usar por ejemplo un objeto Activity, por fuera de acciones de respuesta de una api, tendria esos 3 campos de mas...No se si se entiende, uso any solo para obtener esos tres campos, los demas si los mapeo a la entidad.
  Sino deciem como deberia hacerlo, la entidad Activity tendria los atributos de una actividad y 3 atributos q son de las respuestas http?

Reply:
Sucede que yo veo a la Entidad Activity como una Activity, no como una ActivityHttp por llamarla de una manera, Sino deberia hacerla asi y los metodos devolver httpResponseActivity y no Activity
![image](https://user-images.githubusercontent.com/18562089/139963886-4a085ae4-280a-4fae-8567-128e1ce794d3.png)